src/main/java/gitflow/GitflowVersionTester.java

```diff
@@ -1,31 +1,58 @@
 package gitflow;
 
-import com.intellij.execution.configurations.GeneralCommandLine;
-import com.intellij.execution.ExecutionException;
-import com.intellij.execution.util.ExecUtil;
-import com.intellij.execution.process.ProcessOutput;
+import com.intellij.openapi.components.ServiceManager;
+import com.intellij.openapi.diagnostic.Logger;
+import com.intellij.openapi.project.Project;
+
+import java.util.Map;
+import java.util.concurrent.ConcurrentHashMap;
 
 public class GitflowVersionTester {
-	static Boolean isSupportedVersion = null;
 
-	static boolean isSupportedVersion(){
-		if (isSupportedVersion == null) {
+	private static final Logger logger = Logger.getInstance(GitflowVersionTester.class);
+
+	private static final Map<Project, GitflowVersionTester> testers = new ConcurrentHashMap<>();
+
+	public static GitflowVersionTester forProject(Project project) {
+		// Java 8
```

Comment:
Feel free to use the Java 8 syntax - I switched from 6 a while ago. I'll update the comment in the guide :)

Reply:
Heavens! Yes please!